classes/Link.php

```diff
@@ -820,6 +824,30 @@ public function getAdminLink($controller, $withToken = true, $sfRouteParams = ar
         return $this->getAdminBaseLink() . basename(_PS_ADMIN_DIR_) . '/' . Dispatcher::getInstance()->createUrl($controller, $idLang, $params);
     }
 
+    /**
+     * @param \Symfony\Component\Routing\RouterInterface $sfRouter
+     * @param string                                     $controller
+     *
+     * @return string|null
+     */
+    private function searchRouteFromRouter(\Symfony\Component\Routing\RouterInterface $sfRouter, $controller)
+    {
+        /**
+         * @var string                           $routeName
+         * @var \Symfony\Component\Routing\Route $route
+         */
+        foreach ($sfRouter->getRouteCollection() as $routeName => $route) {
+            if (in_array('GET', $route->getMethods())) {
+                $routeDefaults = $route->getDefaults();
+                if (isset($routeDefaults['_legacy_link']) && $controller == $routeDefaults['_legacy_link']) {
+                    return $routeName;
+                }
+            }
+        }
+
+        return null;
```

Comment:
same here

Reply:
"same" as in the "string|null" comment?

Reply:
yes :) avoid return multiple types (better to throw an exception)